Mathlib/FieldTheory/Minpoly/Field.lean

```diff
@@ -126,19 +126,29 @@ theorem eq_of_irreducible [Nontrivial B] {p : A[X]} (hp1 : Irreducible p)
   · rwa [Polynomial.Monic, leadingCoeff_mul, leadingCoeff_C, mul_inv_cancel]
 #align minpoly.eq_of_irreducible minpoly.eq_of_irreducible
 
+/-- If `y` is the image of `x` under an injective algebra homomorphism, their minimal polynomials
+coincide.
+
+We take `h : y = f x` as an argument because `rw [h]` typically fails
+since `IsIntegral K y` depends on y.
+-/
+theorem eq_of_algHom_eq {K S T : Type _} [Field K] [Ring S] [Ring T] [Algebra K S] [Algebra K T]
+    (f : S →ₐ[K] T) (hf : Function.Injective f) {x : S} {y : T} (hx : IsIntegral K x)
+    (h : y = f x) : minpoly K x = minpoly K y :=
+  minpoly.unique _ _ (minpoly.monic hx)
+    (by rw [h, aeval_algHom_apply, minpoly.aeval, AlgHom.map_zero])
+    (fun q q_monic root_q =>
+      minpoly.min _ _ q_monic (by rwa [h, aeval_algHom_apply, map_eq_zero_iff _ hf] at root_q))
```

Comment:
Is there some overlap here with `minpoly_algHom`?

Reply:
Yep, `rw [h, minpoly_algHom _ hf]` works here!

bors r-

Reply:
Nice, and we only need [CommRing K].

Reply:
It does look like we're missing `minpoly_algebraMap`, which would follow as a special case just as you do in the PR.

Reply:
#7228